### PR TITLE
FIX-#4044: Update module layout with the ModinDataframe

### LINKED ISSUE #4044: Fix documentation after algebra refactor
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?
The module layout in the system architecture doc is now inaccurate thanks to both the major refactor from last year, and the refactor which introduces the `ModinDataframe`.

### PR DESCRIPTION
**Pull request by RehanSD**
Signed-off-by: Rehan Durrani <rehan@ponder.io>

## What do these changes do?
This PR updates the System Architecture documentation page's module layout with the `ModinDataframe` to reflect the changes from #2656.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4044 ? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/flow/modin/core/dataframe/base/dataframe.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+ModinDataframe
+"""""""""""""""
+
+The :py:class:`~modin.core.dataframe.base.dataframe.dataframe.ModinDataframe` is the parent class for all dataframes - regardless of what storage format they are backed by. Its purpose is to define the algebra operators that must be exposed by a dataframe.
+
+This class exposes the dataframe algebra and is meant to be subclassed by all dataframe implementations.
+Descendants of this class implement the algebra, and act as the intermediate level
+between the query compiler and the underlying exection details (e.g. the conforming partition manager). The class provides
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/base/dataframe.rst:8]:
```suggestion
between the query compiler and the underlying execution details (e.g. the conforming partition manager). The class provides
```

File: docs/flow/modin/core/dataframe/base/dataframe.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+a significantly reduced set of operations that can be composed to form any pandas query.
+
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/base/dataframe.rst:10]:
```suggestion
```

File: docs/flow/modin/core/dataframe/base/dataframe.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
+ModinDataframe
+"""""""""""""""
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/base/dataframe.rst:2]:
```suggestion
ModinDataframe
""""""""""""""
```

File: docs/flow/modin/core/dataframe/base/dataframe.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+The :py:class:`~modin.core.dataframe.base.dataframe.dataframe.ModinDataframe` is the parent class for all dataframes - regardless of what storage format they are backed by. Its purpose is to define the algebra operators that must be exposed by a dataframe.
+
+This class exposes the dataframe algebra and is meant to be subclassed by all dataframe implementations.
+Descendants of this class implement the algebra, and act as the intermediate level
+between the query compiler and the underlying exection details (e.g. the conforming partition manager). The class provides
+a significantly reduced set of operations that can be composed to form any pandas query.
+
+
+The ``PandasDataframe`` is an example of a descendant of this class. It currently has implementations for some of the operators
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/base/dataframe.rst:12]:
```suggestion
The :py:class:`~modin.core.dataframe.pandas.dataframe.dataframe.PandasDataframe` is an example of a descendant of this class. It currently has implementations for some of the operators
```

File: docs/flow/modin/core/dataframe/base/dataframe.rst [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+exposed in this class, and is currently being refactored to include implementations for all of the algebra operators. Please
+refer to the :doc: `PandasDataframe documentation<flow/modin/core/dataframe/pandas/dataframe.rst>` for more information.
+
+The ``ModinDataframe`` is independent of implementation specific details such as partitioning, storage format, or execution engine.
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/base/dataframe.rst:16]:
```suggestion
The :py:class:`~modin.core.dataframe.base.dataframe.dataframe.ModinDataframe` is independent of implementation specific details such as partitioning, storage format, or execution engine.
```

**Review by YarShev — APPROVED** [review]
LGTM!

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @RehanSD!

**Pull request merged**